saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/classifier/ConstrainedClassifier.scala

```diff
@@ -132,25 +132,28 @@ abstract class ConstrainedClassifier[T <: AnyRef, HEAD <: AnyRef](val onClassifi
       )(o.asInstanceOf[T])
   }
 
-  /** Test with given data, use internally
+  /** Derives Test Instances from the data model.
     *
-    * @return List of (label, (f1, precision, recall))
+    * @return Iterable of test instances for this classifier
     */
-  def test(): Results = {
-    val allHeads: Iterable[HEAD] = {
-      if (pathToHead.isEmpty) {
+  private def deriveTestInstances: Iterable[T] = {
+    pathToHead.map(edge => edge.from)
```

Comment:
Minor: I usually would say `pathToHead.map(_.from)`, but now seems like keeping it this way would increase the readability.